pysierraecg/src/sierraecg/lib.py

```diff
@@ -107,7 +123,38 @@ def assert_leads(elt: Document) -> List[EcgLead]:
         lead.samples = waveform_data[index]
         leads.append(lead)
 
-    return leads
+    return leads, labels
+    
+def assert_reps(elt, labels):
+    
+    # get repbeats dataencoding
+    rep_encoding = get_attr(get_node(elt, "repbeats"), "dataencoding")
+    rep_sampleingrate = get_attr(get_node(elt, "repbeats"), "samplespersec")
+    rep_resolution = get_attr(get_node(elt, "repbeats"), "resolution")
+    rep_method = get_attr(get_node(elt, "repbeats"), "repbeatmethod")
+    # not using get_nodes becuase cast only returns one waveform
+    rep_repbeats = elt.getElementsByTagName("repbeat")
+    rep_waveforms = elt.getElementsByTagName("waveform")
+    # check leads in same order as parsedwaveform
+    rep_labels = [get_attr(item, "leadname") for item in rep_repbeats]
+    if rep_labels != labels:
+        raise ValueError("Ordering of Leads not same for parsedwaveform and repbeats")
+    repbeats: List[EcgRepbeat] = []
+    for idx, item in enumerate(rep_waveforms):
+        repbeat = EcgRepbeat()
+        duration = get_attr(item, "duration")
+        if rep_encoding == "Base64":
+            rep_waveform_data = read_base64_encoding(get_text(item))
+        rep_waveform_data = np.frombuffer(rep_waveform_data, dtype= np.uint16)
```

Comment:
I think this change will get you the right values:
```suggestion
        rep_waveform_data = np.frombuffer(rep_waveform_data, dtype= np.int16)
```
The repbeats are signed 16-bit samples rather than unsigned.

Reply:
This worked great!